test/tests/selftest/pbs_pbstestsuite.py

```diff
@@ -0,0 +1,132 @@
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.selftest import *
+
+
+class TestPBSTestSuite(TestSelf):
+    """
+    Contains tests for pbs_testsuite module's functionality
+    """
+
+    def test_revert_pbsconf_onehost(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf()
+        for a single host type 1 installation
+        """
+        pbs_conf_filepath = self.du.get_pbs_conf_file(self.server.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname,
+                                                pbs_conf_filepath)
+        if pbs_conf_val is None or len(pbs_conf_val) < 1:
+            raise ValueError("Could not parse pbs.conf on host %s" %
+                             (server.hostname))
+
+        # Since the setUp already ran, check that the start bits are turned on
+        self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_SERVER"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_SCHED"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_COMM"], "1")
+
+        self.server.pi.stop()
+
+        # Now, change pbs.conf to turn the sched off
+        pbs_conf_val["PBS_START_SCHED"] = "0"
+        self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
+
+        # Start PBS again
+        self.server.pi.start()
+
+        # Verify that the scheduler didn't come up
+        self.assertFalse(self.scheduler.isUp())
+
+        # Now call revert_pbsconf() & restart PBS again
+        self.server.pi.stop()
+        self.revert_pbsconf()
+        self.server.pi.start()
```

Comment:
Do you have to do this? isn't revert_pbsconf starting sched daemon automatically?